netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -205,6 +209,18 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
     }
   }
 
+  private String getOrUpdateAuthority(AsciiString authority) {
+    if (authority == null) {
+      return null;
+    } else if (!authority.equals(lastKnownAuthority)) {
+      lastKnownAuthority = authority;
+    }
+
+    // AsciiString.toString() is internally cached, so subsequent calls will not
```

Comment:
getAuthority() is always called.

This is plenty efficient enough. It doesn't hinder the common case at all.

Most clients, by far, won't change the authority every request. And even if they do change constantly, then any optimization here would probably be in the noise.

I'd rather get to a point we can call good enough for now and any future optimization for edge cases gets included with the HPACK tricks we're planning.